### ISSUE TITLE
 "Request failed" - Slotlist no longer working

### ISSUE THREAD

**Issue by XamGe** [MorpheusXAUT/slotlist-frontend#155]
Slotlist-Page does not load the calendar and scolling down is not possible. The errormessage is:

 "Failed to load missions for calendar - Request failed"

My Slotlist-Username is: [TacNet] Max

**Comment by MorpheusXAUT** [comment]
Just tried multiple systems, browsers and different locations all around Europe and couldn't reproduce the issue. Can you try logging out in case you're logged in or in an incognito tab to see if it still persists?
If it does, could you open the network tab of your browser (should be `CTRL`+`Shift`+`I` on all major browsers) and attach a screenshot/log of the failed requests/responses?

**Comment by XamGe** [comment]
When logged out, the error does not occur. I tried using edge or firefox, clearing cookies, incognito-mode and my andoid phone. 

I think u were asking for this? It is repeating again and again and again nonstop, als long as i am on the website.

`response error https://api.slotlist.info/v1/missions?startDate=1732489200000&endDate=1736117999999 Error: Network Error
    exports createError.js:16
    onerror xhr.js:87
    exports xhr.js:84
    _execute bluebird.js:971
    _resolveFromExecutor bluebird.js:3249
    r bluebird.js:2845
    exports xhr.js:12
    exports dispatchRequest.js:52
    r bluebird.js:5256
    _settlePromiseFromHandler bluebird.js:3278
    _settlePromise bluebird.js:3335
    _settlePromise0 bluebird.js:3380
    _settlePromises bluebird.js:3459
    _drainQueue bluebird.js:185
    _drainQueues bluebird.js:195
    drainQueues bluebird.js:69
    promise callback*o bluebird.js:4484
    _queueTick bluebird.js:204
    o bluebird.js:135
    _then bluebird.js:3027
    then bluebird.js:2891
    request Axios.js:58
    e Axios.js:68
    exports bind.js:9
    getMissionsForCalendar missions.js:78
    getMissionsForCalendar missions.js:2107
    d vuex.esm.js:598
    dispatch vuex.esm.js:341
    dispatch vuex.esm.js:266
    weeks client.7c9dff04.js:1963
    VueJS 3
    render CalendarBody.vue:13
    VueJS 13
    refreshingMissionsForCalendar missions.js:2840
    g vuex.esm.js:591
    commit vuex.esm.js:315
    commit vuex.esm.js:314
    _withCommit vuex.esm.js:402
    commit vuex.esm.js:313
    commit vuex.esm.js:269
    getMissionsForCalendar missions.js:2144
    r bluebird.js:5256
    _settlePromiseFromHandler bluebird.js:3278
    _settlePromise bluebird.js:3335
    _settlePromise0 bluebird.js:3380
    _settlePromises bluebird.js:3455
    _drainQueue bluebird.js:185
    _drainQueues bluebird.js:195
    drainQueues bluebird.js:69
    promise callback*o bluebird.js:4484
    _queueTick bluebird.js:204
    c bluebird.js:140
    _reject bluebird.js:3422
    _rejectCallback bluebird.js:3240
    t bluebird.js:3252
    onerror xhr.js:87
[util.js:58:10](https://slotlist.info/app/src/api/util.js)
getMissionsForCalendar 
XMLHttpRequest { readyState: 4, timeout: 0, withCredentials: false, upload: XMLHttpRequestUpload, responseURL: "", status: 0, statusText: "", responseType: "", response: "", responseText: "" }
[missions.js:2158:18](https://slotlist.info/app/src/store/modules/missions.js)
request https://api.slotlist.info/v1/missions?startDate=1732489200000&endDate=1736117999999 
Object { adapter: exports(e), transformRequest: {…}, transformResponse: {…}, timeout: 0, xsrfCookieName: "XSRF-TOKEN", xsrfHeaderName: "X-XSRF-TOKEN", maxContentLength: -1, validateStatus: validateStatus(e)
, headers: {…}, baseURL: "https://api.slotlist.info/", … }
[util.js:29:10](https://slotlist.info/app/src/api/util.js)
XHRGET
https://api.slotlist.info/v1/missions?startDate=1732489200000&endDate=1736117999999
CORS Missing Allow Origin

Quellübergreifende (Cross-Origin) Anfrage blockiert: Die Gleiche-Quelle-Regel verbietet das Lesen der externen Ressource auf https://api.slotlist.info/v1/missions?startDate=1732489200000&endDate=1736117999999. (Grund: CORS-Kopfzeile 'Access-Control-Allow-Origin' fehlt). Statuscode: 520.

response error https://api.slotlist.info/v1/missions?startDate=1732489200000&endDate=1736117999999 Error: Network Error
    exports createError.js:16
    onerror xhr.js:87
    exports xhr.js:84
    _execute bluebird.js:971
    _resolveFromExecutor bluebird.js:3249
    r bluebird.js:2845
    exports xhr.js:12
    exports dispatchRequest.js:52
    r bluebird.js:5256
    _settlePromiseFromHandler bluebird.js:3278
    _settlePromise bluebird.js:3335
    _settlePromise0 bluebird.js:3380
    _settlePromises bluebird.js:3459
    _drainQueue bluebird.js:185
    _drainQueues bluebird.js:195
    drainQueues bluebird.js:69
    promise callback*o bluebird.js:4484
    _queueTick bluebird.js:204
    o bluebird.js:135
    _then bluebird.js:3027
    then bluebird.js:2891
    request Axios.js:58
    e Axios.js:68
    exports bind.js:9
    getMissionsForCalendar missions.js:78
    getMissionsForCalendar missions.js:2107
    d vuex.esm.js:598
    dispatch vuex.esm.js:341
    dispatch vuex.esm.js:266
    weeks client.7c9dff04.js:1963
    VueJS 3
    render CalendarBody.vue:13
    VueJS 12`

**Comment by MorpheusXAUT** [comment]
I think I've managed to narrow down the issue a bit more, although I'm not 100% sure yet.

Shoddy database design and programming eight years ago paired with a tiny server to keep operating costs low basically means with the 100+ missions you've created, 275+ permissions granted and over 7000 notifications stored for your user, the database and backend are simply timing out sometimes when trying to load your user on initial page load (which requests basically all data combined).

Long story short: I'll have to figure out how to "fix" this with minimal effort as I certainly can't optimise the backend or database much and would rather not have to upgrade to a larger server.
One option might be to delete some of the oldest data associated with your user if you're fine with that, as that should alleviate the load a bit...

**Comment by XamGe** [comment]
Please feel free to delete any old data. Thanks for ur fast help and much effort keeping slotlist alive.

**Comment by XamGe** [comment]
Moin & Greetings. The slotlist-user Wilpie seems to have the same issue as I had. Would you kindley delete some old data for him too?

**Comment by MorpheusXAUT** [comment]
Hi @XamGe, sorry for the delay. I've removed all the seen "notifications" as well as all mission permissions older than 6 months for both you and Wilpie. In a quick test, this allowed me to load data again - could you please test?
The missions themselves have not been touched, so the data is still there, you might just lack permission to access/modify it. If you need to re-gain access to some of these, please let me know the exact ones and I'll restore them by hand.
